File: 프로그래머스/2/42583. 다리를 지나는 트럭/다리를 지나는 트럭.js
function solution(bridge_length, weight, truck_weights) {
  let time = 0;
  const bridge = new Array(bridge_length).fill(0);
  let current_weight = 0;

  while (bridge.length) {
    time += 1;
    current_weight -= bridge.shift();

    // 주석 추가
    if (truck_weights.length) {
      if (current_weight + truck_weights[0] <= weight) {
        const next_truck = truck_weights.shift();
        bridge.push(next_truck);
        current_weight += next_truck;
      } else {
        bridge.push(0);
      }
    }
  }

  return time;
}

File: 프로그래머스/2/42583. 다리를 지나는 트럭/다리를 지나는 트럭.py
def solution(bridge_length, weight, truck_weights):
    time = 0
    bridge = [0] * bridge_length
    current_weight = 0
    
    while bridge:
        time += 1
        current_weight -= bridge.pop(0)
        
        if truck_weights:
            if current_weight + truck_weights[0] <= weight:
                next_truck = truck_weights.pop(0)
                bridge.append(next_truck)
                current_weight += next_truck
            else:
                bridge.append(0)
    
    return time
